wolfcrypt/src/port/maxim/maxq10xx.c

```diff
@@ -0,0 +1,3363 @@
+    }
+
+    int obj_id = alloc_aes_key_id();
+    if (!obj_id) {
+        WOLFSSL_ERROR_MSG("MAXQ: alloc_aes_key_id() failed");
+        rc = NOT_COMPILED_IN;
+        goto end_AesSetKey;
+    }
+
+    mxq_rc = MXQ_CreateObject(obj_id, keylen, MXQ_OBJTYPE_SECRETKEY,
+                              OBJPROP_PERSISTENT,
+                              (char *)"ahs=rwdgx:ahs=rwdgx:ahs=rwdgx");
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_CreateObject() failed");
+        rc = NOT_COMPILED_IN;
+        goto end_AesSetKey;
+    }
+
+    /* store the object id in the context */
+    aes->maxq_ctx.key_obj_id = obj_id;
+
+    mxq_length key_buff_len = sizeof(key_buff);
+    mxq_rc = MXQ_BuildKey(key_buff, &key_buff_len, MXQ_KEYTYPE_AES, 0xff,
+                          keylen, keylen, MXQ_KEYUSE_ENCRYPTION,
+                          ALGO_CIPHER_AES_ECB, MXQ_KEYUSE_NONE, ALGO_NONE,
+                          (mxq_u1 *)userKey);
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_BuildKey() failed");
+        rc = WC_HW_E;
+        goto end_AesSetKey;
+    }
+
+    HOST_LoadDefaultImportKey(sign_key, &sign_key_len, &sign_key_curve,
+                              &sign_key_type);
+
+    rc = HOST_ECDSA_sign(signature, &signature_len, sign_key,
+                key_buff, key_buff_len, sign_key_curve);
+    if (rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: HOST_ECDSA_sign() failed");
+        goto end_AesSetKey;
+    }
+
+    mxq_rc = MXQ_ImportKey(obj_id, getSignAlgoFromCurve(sign_key_curve),
+                           PUBKEY_IMPORT_OBJID, key_buff, key_buff_len,
+                           signature, signature_len);
+    if (mxq_rc) {
+        WOLFSSL_ERROR_MSG("MAXQ: MXQ_ImportKey() failed");
+        rc = WC_HW_E;
+        goto end_AesSetKey;
+    }
+
+    /* key stored successfully */
+    aes->maxq_ctx.key_pending = 0;
+
+end_AesSetKey:
+    wolfSSL_CryptHwMutexUnLock();
+    return rc;
+}
+#endif /* MAXQ_AESGCM */
+
+void wc_MAXQ10XX_AesFree(Aes* aes)
+{
+    mxq_err_t mxq_rc;
+    int rc = 1;
+
+    if (aes->maxq_ctx.key_obj_id != 0) {
+        rc = 0;
+    }
+
+    if (rc == 0) {
+        rc = wolfSSL_CryptHwMutexLock();
+    }
+
+    if (rc == 0) {
+        mxq_rc = MXQ_DeleteObject(aes->maxq_ctx.key_obj_id);
+        if (mxq_rc) {
+            WOLFSSL_ERROR_MSG("MAXQ: MXQ_DeleteObject() failed");
+            rc = 1;
+        }
+
+        if (rc == 0) {
+            free_aes_key_id(aes->maxq_ctx.key_obj_id);
+            aes->maxq_ctx.key_obj_id = 0;
+        }
+        wolfSSL_CryptHwMutexUnLock();
+    }
+}
+
+#ifdef MAXQ_ECC
+static int ecc_set_key(ecc_key* key, const byte* userKey, word32 keycomplen)
+{
+    mxq_err_t mxq_rc;
+    int rc;
+    word32 keylen;
+    int objtype;
+    mxq_u1 key_buff[256];
+    mxq_length key_buff_len = sizeof(key_buff);
+    unsigned char sign_key[96];
+    int sign_key_len, sign_key_curve, sign_key_type;
+    mxq_u1 signature[64];
+    int signature_len = sizeof(signature);
```

Comment:
Cast sizeof to `int`.

Reply:
fixed with d728e4c67c62e9fb38a7c44e8ce75e018e685bf3